doc/api/plotting/charts/index.rst

```diff
@@ -0,0 +1,25 @@
+Charts
+------
+Charts API reference. These dedicated classes can be used to embed
+charts in plotting windows.
+
+.. currentmodule:: pyvista
+
+.. autosummary::
+   :toctree: _autosummary
+   :template: custom-class-template.rst
+
+   plotting.charts.Pen
+   plotting.charts.Brush
+   plotting.charts.Axis
+   plotting.charts.LinePlot2D
+   plotting.charts.ScatterPlot2D
+   plotting.charts.BarPlot
+   plotting.charts.AreaPlot
+   plotting.charts.StackPlot
+   Chart2D
+   plotting.charts.BoxPlot
+   ChartBox
+   plotting.charts.PiePlot
+   ChartPie
+   ChartMPL
```

Comment:
I haven't had time to run this PR's build yet. How do these guys get rendered? I wonder if we should separate them according to "pulled into main `pyvista` namespace" vs "`pyvista.charts` only".

Reply:
Yes that's possible, the current order is just the order in `charts.py`. I will add a screenshot of the docs here tomorrow.

Reply:
Yes that's possible, the current order is just the order of definition in `charts.py`. I will add a screenshot of the docs here.

Reply:
Seems something is broken in the sidebar of the docs, will investigate that tomorrow.
Charts overview page:
![image](https://user-images.githubusercontent.com/15089458/143132347-fca1aa11-ab0f-45b2-8f16-b62686c81bb5.png)
Pen class (sidebar is gone here):
![image](https://user-images.githubusercontent.com/15089458/143132426-7d858b2d-5767-481b-9a39-19bf4bb74adf.png)

Reply:
Awesome, thank you.

So it's clear that the namespace differences are obvious on the main Charts page. I'd pull the four exceptions to the front, and maybe even write a paragraph before the class list that most of the chart classes are in `pyvista.plotting.charts`, with the exceptions of [the classes] for convenience reasons. Or something along those lines.

We'd want people to use the short names where available, but at the same time they should know that there's treasure two namespaces deeper.

Reply:
Ok, will do that. As an extra, do you know why sphinx uses `pyvista.plotting.charts.[class]` while the correct way to access these classes in python is `pyvista.charts.[class]` and is there a way to make this consistent?

Reply:
I'm afraid I'm a sphinx/numpydoc noob, so I don't know how to fix that automatically. You can probably define them manually with a short name.

A related example that comes to mind is [`pv.Light`](https://dev.pyvista.org/api/core/lights.html):
```py
>>> import pyvista as pv
>>> type(pv.Light())
<class 'pyvista.plotting.lights.Light'>
```

And its API reference resolves
```
:class:`pyvista.Light`
```
just fine. But that case might be a bit different, because we don't have such a list of explicit classes for Lights...

Reply:
I just pushed part 2 of the changes, this should have addressed your remaining suggestions. The docs are also correctly building now.